content/modules/ROOT/pages/publish-an-addon.adoc

```diff
@@ -0,0 +1,173 @@
+            maven {
+                name = "GitHubPackages"
+                url = uri("https://maven.pkg.github.com/" + System.getenv("GITHUB_REPOSITORY"))
+                credentials {
+                    username = project.findProperty("gpr.user") ?: System.getenv("GITHUB_USERNAME")
+                    password = project.findProperty("gpr.key") ?: System.getenv("GITHUB_TOKEN")
+                }
+            }
+        }
+        publications {
+            javaMaven(MavenPublication) {
+                artifactId = archName
+                from components.java
+            }
+        }
+    }
+----
+
+Additionally, you should remove the version attribute in `build.gradle` (under `group = 'io.jmix.my.new.addon'`). Instead, create a file called `gradle.properties` with the version information directly next to the `build.gradle` file:
+
+[source,properties]
+.gradle.properties
+----
+version=0.0.1-SNAPSHOT
+----
+
+
+== Continuous Delivery
+
+GitHub offers a built-in CI system, that compiles the code, runs tests, performs releases, etc. for free for open-source projects.
+
+By adding the following GitHub action workflow files into your GitHub repository, the CI system will perform compilation and execution of automated tests. You will need to create the directory `.github/workflows` manually in the repository.
+
+[source,yaml]
+.test.yml
+----
+name: CI pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  ci:
+    name: CI pipeline
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v1
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Run tests
+        run: ./gradlew test
+
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v2
+        if: always() # always run even if the previous step fails
+        with:
+          report_paths: '**/build/test-results/test/TEST-*.xml'
+----
+
+[source,yaml]
+.release.yml
+----
+name: Publish release
+on:
+  release:
+    types: [created]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Publish package
+        run: gradle -Pversion=${{ github.event.release.tag_name }} build publish
+        env:
+          GITHUB_REPOSITORY: ${{ github.repository }}
+          GITHUB_USERNAME: ${{ github.repository_owner }}
+          GITHUB_TOKEN: ${{ github.token }}
+----
+
+With those two files in place GitHub Actions will perform the following tasks:
+
+* compile the code
+* run unit / integration tests
+* store test results
+* publishes a new version for newly created releases
+
+== Register Addon in Marketplace
+
+// TODO: how to register a new addon in the marketplace?
+To publish an addon to the https://www.jmix.io/marketplace/[Jmix Marketplace], you need to ... (TBD).
```

Comment:
Perhaps it will depend on our decision about jmix-marketplace repository?

Reply:
Right, I left it as a todo for now since we hadn’t talked about the way it will work. Let’s see what @SofiaOganova  says about it and then I will adjust it accordingly.